crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -345,6 +353,76 @@ unsafe impl QueryData for Entity {
 /// SAFETY: access is read only
 unsafe impl ReadOnlyQueryData for Entity {}
 
+/// SAFETY:
+/// `update_component_access` and `update_archetype_component_access` do nothing.
+/// This is sound because `fetch` does not access components.
+unsafe impl WorldQuery for EntityLocation {
+    type Item<'w> = EntityLocation;
+    type Fetch<'w> = &'w Entities;
+    type State = ();
+
+    fn shrink<'wlong: 'wshort, 'wshort>(item: Self::Item<'wlong>) -> Self::Item<'wshort> {
+        item
+    }
+
+    unsafe fn init_fetch<'w>(
+        world: UnsafeWorldCell<'w>,
+        _state: &Self::State,
+        _last_run: Tick,
+        _this_run: Tick,
+    ) -> Self::Fetch<'w> {
+        world.entities()
+    }
+
+    const IS_DENSE: bool = true;
```

Comment:
So the reason for it to be true is so that if for tuple queries (`Query<(A, B)>`), we want the final QueryState::IS_DENSE (equal to `A::State::IS_DENSE && B::State::IS_DENSE`) to be true for optimization?

If that's the case, I would maybe add a comment

Reply:
Done.